Dockerfile

```diff
@@ -18,12 +18,15 @@ ARG VALIDATION=OFF
 # This creates a release build that includes link time optimization and links
 # all libraries statically.
 ARG BUILD_STATIC_RELEASE=OFF
+# Toggle usage of the GNU Multiple Precision Arithmetic Library.
+ARG ENABLE_GMP=ON
 # No questions asked during package installation.
 ARG DEBIAN_FRONTEND=noninteractive
 # Whether to install dependencies required to run PTF-ebpf tests
 ARG INSTALL_PTF_EBPF_DEPENDENCIES=OFF
 
 # Delegate the build to tools/ci-build.
 COPY . /p4c/
+RUN --mount=type=bind,source=.ccache,target=/root/.ccache,rw  chmod u+x /p4c/tools/ci-build.sh && /p4c/tools/ci-build.sh
```

Comment:
@fruffy it might be better to add the `--mount` option to the `docker run` command instead because we also use this Dockerfile to build the [p4lang/p4c](https://hub.docker.com/r/p4lang/p4c) container image in Dockerhub.

Reply:
So the problem is that the `docker run` command is just running ctest. However, we need to map ccache correctly when building p4c. I am not sure how to untangle this yet.

Reply:
We could use something like the following:
```diff
-RUN chmod u+x tools/ci-build.sh && tools/ci-build.sh
+RUN (test "$IMAGE_TYPE" = "build" && \
+        chmod u+x tools/ci-build.sh && tools/ci-build.sh) || \
+    (test "$IMAGE_TYPE" = "test" && \
+      echo 'Test image ready')
```

and

```yaml
    - name: Build (Linux)
      run: |
        docker build -t p4c --build-arg IMAGE_TYPE=test .
        docker run --mount type=bind,source=/usr/lib/ccache,target=/root/.ccache -e ENABLE_UNIFIED_COMPILATION=ON -e BUILD_STATIC_RELEASE=ON -e ENABLE_GMP=ON p4c bash -c 'chmod u+x tools/ci-build.sh && tools/ci-build.sh'

```

Reply:
Does the current solution work for you?

Reply:
Yes

Reply:
Yes, I tested it locally and it seems to work.